tests/kafkatest/services/verifiable_producer.py

```diff
@@ -68,29 +68,33 @@ def __init__(self, context, num_nodes, kafka, topic, max_messages=-1, throughput
         self.message_validator = message_validator
         self.compression_types = compression_types
 
+        self.security_config = self.kafka.security_config.client_config()
+
         for node in self.nodes:
             node.version = version
         self.acked_values = []
         self.not_acked_values = []
         self.produced_count = {}
-        self.prop_file = ""
+
+    def prop_file(self, node):
+        idx = self.idx(node)
+        prop_file = str(self.security_config)
+        if self.compression_types is not None:
+            compression_index = (idx - 1) % len(self.compression_types)
```

Comment:
On third thought, this maybe doesn't really come up and therefore is a pretty minor point - but unless there's a good reason to have the modulo, it seems unnecessarily tricky?